python/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+if(GTSAM_EXAMPLE_BUILD_PYTHON)
+  set(PYBIND_WRAP_MODULE_NAME gtsam_example)
+  
+  # Generate setup.py.
+  file(READ "${PROJECT_SOURCE_DIR}/README.md" README_CONTENTS)
+  configure_file(${PROJECT_SOURCE_DIR}/python/setup.py.in
+                 ${PROJECT_BINARY_DIR}/setup.py)
+
+  add_subdirectory(pybind11)
+  include(GtsamPybindWrap)
+
+  add_custom_target(${PYBIND_WRAP_MODULE_NAME}_header DEPENDS "${PROJECT_SOURCE_DIR}/${PYBIND_WRAP_MODULE_NAME}.h")
+
+  # ignoring the non-concrete types (type aliases)
+  set(ignore
+      )
+
+  pybind_wrap(${PYBIND_WRAP_MODULE_NAME}_py # target
+            ${PROJECT_SOURCE_DIR}/${PYBIND_WRAP_MODULE_NAME}.h # interface_header
+            "${PYBIND_WRAP_MODULE_NAME}.cpp" # generated_cpp
+            "${PYBIND_WRAP_MODULE_NAME}" # module_name
+            "${PYBIND_WRAP_MODULE_NAME}" # top_namespace
+            "${ignore}" # ignore_classes
+            ${PROJECT_SOURCE_DIR}/wrap/pybind_wrapper.tpl
+            ${PYBIND_WRAP_MODULE_NAME} # libs
+            "${PYBIND_WRAP_MODULE_NAME};${PYBIND_WRAP_MODULE_NAME}_header" # dependencies
+            )
+
+  set_target_properties(${PYBIND_WRAP_MODULE_NAME}_py PROPERTIES
+        INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib"
```

Comment:
Don't need some of these since we do a python install, but I have addressed them in my PR.